cirq/ops/pauli_string.py

```diff
@@ -163,27 +164,52 @@ def _value_equality_values_cls_(self):
             return gate_operation.GateOperation
         return PauliString
 
-    def equal_up_to_coefficient(self, other: 'PauliString') -> bool:
+    def equal_up_to_coefficient(self, other: 'cirq.PauliString') -> bool:
         return self._qubit_pauli_map == other._qubit_pauli_map
 
-    def __getitem__(self, key: 'cirq.Qid') -> pauli_gates.Pauli:
+    def __getitem__(self, key: TKey) -> pauli_gates.Pauli:
         return self._qubit_pauli_map[key]
 
     # pylint: disable=function-redefined
     @overload
-    def get(self, key: 'cirq.Qid', default: None = None) -> pauli_gates.Pauli:
+    def get(self, key: Any, default: None = None) -> pauli_gates.Pauli:
         pass
 
     @overload
-    def get(self, key: 'cirq.Qid',
+    def get(self, key: Any,
             default: TDefault) -> Union[pauli_gates.Pauli, TDefault]:
         pass
 
-    def get(self, key: 'cirq.Qid', default=None):
+    def get(self, key: Any, default=None):
         return self._qubit_pauli_map.get(key, default)
-    # pylint: enable=function-redefined
 
-    def __mul__(self, other) -> 'PauliString':
+    @overload
+    def __mul__(  # type: ignore
+            self, other: 'cirq.PauliString[TKey2]'
+    ) -> 'cirq.PauliString[Union[TKey, TKey2]]':
+        pass
+
+    @overload
+    def __mul__(self, other: Mapping[TKey2, 'cirq.PAULI_GATE_LIKE']
+               ) -> 'cirq.PauliString[Union[TKey, TKey2]]':
```

Comment:
this Union could be pulled out as a type alias for "AllKeyTypes" or similar?

Reply:
It doesn't make sense to define that constant, because it is used few times and also the meaning of `TKey2` is method specific (in the sense that it is "bound" separately by each method). `Union[TKey, TKey2]` really doesn't mean "all key types", it means "the result's keys come from either this pauli string or the given mapping".

Reply:
Thanks for explaining. I get it now. TKey2 is used to allow for interactions with other PauliStrings that will be keyed potentially with a different type.